### ISSUE TITLE
Ctrl+Z (undo) not working any more

### ISSUE THREAD

**Issue by koppor** [JabRef/jabref#11420]
Originally reported at https://github.com/JabRef/jabref/pull/11282#issuecomment-2177466526

An exception is thrown when trying to use <kbd>Ctrl</kbd>+<kbd>Z</kbd> on any text field in the entry editor.

```
2024-06-19 06:13:03 [JavaFX Application Thread] org.jabref.gui.FallbackExceptionHandler.uncaughtException()
ERROR: Uncaught exception occurred in Thread[#54,JavaFX Application Thread,5,main]: java.lang.NullPointerException: Cannot read field "next" because "this.undoChange" is null
	at javafx.controls@22.0.1/javafx.scene.control.TextInputControl.updateUndoRedoState(TextInputControl.java:1250)
```

A work-around is to disable the key combinations Ctrl+Z and Ctrrl+Y for the field:

```java
        // We need to consume the key event to avoid the default behavior of undo/redo and enable JabRef's undo/redo
        // Source: https://stackoverflow.com/a/37575818/873282
        textInputControl.addEventFilter(KeyEvent.ANY, e -> {
            if (e.getCode() == KeyCode.Z && e.isShortcutDown()) {
                e.consume();
            }
        });
```

However, the undo/redo should still work. I went through the code a bit, but `org.jabref.gui.frame.JabRefFrame.UndoRedoEventManager#updateTexts` is "dead" code and I don't know further (yet).

**Comment by koppor** [comment]
In parallel to a fix, we should create an MWE showing the error inside (!) JavaFX and report a bug.

**Comment by Siedlerchr** [comment]
I think when previously the field was empty, and you now enter something 
ctrl + z
exception occurs

Now type again and ctrl + z again it works fine

**Comment by ryan-carpenter** [comment]
> I think when previously the field was empty, and you now enter something ctrl + z exception occurs
> 
> Now type again and ctrl + z again it works fine



> I think when previously the field was empty, and you now enter something ctrl + z exception occurs
> 
> Now type again and ctrl + z again it works fine

The exception also occurs when there is text in the field.

**Comment by koppor** [comment]
I think, it was introduced with our bindings

https://github.com/JabRef/jabref/pull/11282/files#diff-12c8d5c64666483ab38360768d084135c872d59c6798fe13184dfdec888d99eb

As said at https://github.com/JabRef/jabref/pull/11421#issuecomment-2186841645: New reproducer (I started to work on one but did not push 🙈)

**Comment by LoayGhreeb** [comment]
Using the menu bar (Edit -> Undo) to undo changes works without any exceptions.

Another issue, which may or may not be related, is that when updating a field using the context menu (Clear, Change Case, Normalize) and using <kbd>Ctrl</kbd>+<kbd>Z</kbd> to undo this change, nothing happens and no exception occurs. I can only undo this change from the menu bar. However, when using "Protect Selection" from the context menu, this change is undoable but with the exception.

**Comment by koppor** [comment]
JavaFX TextField issue: Asked in #java-german (https://jvm-german.slack.com/archives/C4H2FQKND/p1720297099980919) and created an MWE: https://github.com/Siedlerchr/javafxreproducer/pull/22

**Comment by koppor** [comment]
I checked the source code at https://github.com/openjdk/jfx17u/blob/a205815f99903a1a55c3fbb65a904b20584fb3fc/modules/javafx.controls/src/main/java/javafx/scene/control/TextInputControl.java#L1482 - it should work. However, if I decompile TextInputControll, I get a different source. -- Currently investigating.

**Comment by koppor** [comment]
OK, Loay is asking for ["Objektbezogenes Undo"](http://www.usabilitypatterns.info/catalog/patterns/objektundo.html). The text there does not fully explain how it works with hierarchical objects. It also does not link real-world examples (which makes it a "pattern candidate")
